modules/autocomplete-feed/main/index.coffee

```diff
@@ -0,0 +1,99 @@
+    else if a.disabled and not b.disabled then 1
+    else hx.sort.compare(valueLookup(a), valueLookup(b))
+
+trimTrailingSpaces = (term) ->
+  newTerm = term
+  while newTerm.lastIndexOf(' ') is newTerm.length - 1
+    newTerm = newTerm.slice(0, newTerm.length - 1)
+  newTerm
+
+class AutocompleteFeed
+  constructor: (options = {}) ->
+    self = this
+
+    defaults =
+      filter: undefined
+      filterOptions: undefined
+      matchType: 'contains'
+      showOtherResults: false
+      trimTrailingSpaces: false
+      valueLookup: undefined
+
+    if options.valueLookup?
+      # default searchValue if valueLookup is defined
+      defaults.filterOptions =
+        searchValues: (datum) -> [self._.options.valueLookup(datum)]
+
+    resolvedOptions = hx.merge.defined defaults, options
+
+    # defined here so we can use the resolved options
+    resolvedOptions.filter ?= (items, term) ->
+      hx.filter[resolvedOptions.matchType](items, term, resolvedOptions.filterOptions)
+        .sort sortItems(resolvedOptions.valueLookup)
+
+    @_ =
+      options: resolvedOptions
+      resultsCache: new hx.Map
+
+
+  clearCache: -> @_.resultsCache = new hx.Map
+
+  filter: (term = '', callback) ->
+    _ = @_
+
+    cacheitemsAndCallback = (results, otherResults = []) =>
+      if _.options.trimTrailingSpaces and results.length is 0 and term.lastIndexOf(' ') is term.length - 1
+        # The term has trailing spaces and no results were found
+        @filter(trimTrailingSpaces(term), callback)
+      else
+        # Cache the currently searched term items
+        _.resultsCache.set(term, {
+          results: results,
+          otherResults: otherResults
+        })
+        callback(results, otherResults)
+
+    if _.options.matchType is 'external' and hx.isFunction(_.items)
+      # The matching is external so we don't filter here
+      _.items(term, cacheitemsAndCallback)
+    else
+      if _.resultsCache.has(term)
+        # Use the cached items if it exists
+        cacheditems = _.resultsCache.get(term)
+        callback(cacheditems.results, cacheditems.otherResults)
+      else
+        filterAndCallback = (unfilteredItems) ->
+          filteredItems = _.options.filter(unfilteredItems, term)
+          if _.options.showOtherResults
+            otherResults = unfilteredItems.filter (datum) ->
+                filteredItems.indexOf(datum) is -1
+              .sort sortItems(_.options.valueLookup)
+
+          cacheitemsAndCallback(filteredItems, otherResults)
+
+        if hx.isFunction(_.items)
+          # Call the function then apply filtering
+          _.items(term, filterAndCallback)
+        else if term.length
+          # Apply filtering to the static object
+          filterAndCallback(_.items)
+        else
+          # Skip filtering and return the entire itemsset
+          cacheitemsAndCallback(_.items)
+
+  validateItems: (items) -> (hx.isArray(items) and items.length > 0) or hx.isFunction(items)
```

Comment:
The situation I was thinking of was: if you had something else that changes the values available to choose from in the picker, and you only have a small number of items, it could make sense in that case to set the items directly, rather than setting up a function feed. And in that case you _might_ sometimes want to set the items to an empty array. Its a bit contrived I admit, but the use case could happen